packages/babel-helpers/src/helpers.ts

```diff
@@ -788,7 +788,13 @@ helpers.iterableToArrayLimit = helper("7.0.0-beta.0")`
     var _d = false;
     var _s, _e;
     try {
-      for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) {
+      _i = _i.call(arr);
+      if (i === 0) {
+        if (typeof _i !== "object" && typeof _i !== "function") return;
+        _n = false;
+        return _arr;
+      }
+      for (; !(_n = (_s = _i.next()).done); _n = true) {
```

Comment:
I'm from mobile so this code might be completely wrong, but does it work?

```js
for (_i = _i.call(arr); i < _arr.length && !(_n = (_s = _i.next()).done); _n = true) {
  _arr.push(_s.value);
}
```

We should try keeping these helpers as small as possible :)

Reply:
I change it to this:

```js
try {
  for (_i = _i.call(arr); i < _arr.length && !(_n = (_s = _i.next()).done); _n = true) {
    _arr.push(_s.value);
  }
  if (i === 0) {
    if (Object(_i) !== _i) return;
    _n = false;
  }
} // ...
```

Does this looks good?

Reply:
Oh, that seems to be not working since the order of `_n = true` is changed... I'll revert the change.

```js
try {
  _i = _i.call(arr);
  if (i === 0) {
    if (Object(_i) !== _i) return;
    _n = false;
  } else {
    for (; !(_n = (_s = _i.next()).done); _n = true) {
      _arr.push(_s.value);
      if(_arr.length === i) break;
    }
  }
} // ...
```